### DISCUSSION TITLE
How do I call rpc methods in golang

### DISCUSSION THREAD

**Discussion by ryan053857** [roadrunner-server/roadrunner#2041] [Q&A] [answered]
I am integrating roadrunner into my golang application to develop the application. Now I have a scenario: there is an http interface with a very large amount of concurrent traffic, and I want to take this part of traffic in golang, and then forward it to php for processing after some processing. Since the traffic is very large and there is no need to return a response, only the data needs to be processed, in order not to use the http worker, I want to directly send the message to the php queue worker for execution.
How to achieve this? Is it the push method of the rpc in golang that creates an rpc connection to golang and then requests jobs? Can I do this by directly accessing the jobs plugin's methods?

**Comment by rustatian** [comment, marked as answer]
Hey @shellphy 👋 
You may create a middleware, intercept a request, process, and return an empty response or do that in a separate goroutine. Here is a tutorial on how to write a middleware: https://docs.roadrunner.dev/docs/customization/middleware

**Reply by ryan053857** [reply]
But how do you push tasks to a queue in Go? There is no specific example in the documentation.

**Reply by rustatian** [reply]
I'm not sure what queue do you mean. Could you please elaborate?

**Reply by ryan053857** [reply]
And the queue is jobs. I want to push jobs data in golang

Also, I store some data with rr key-value plugin by php, how do I access it in golang?

**Reply by rustatian** [reply]
You can take a look at the roadrunner-php/jobs package. The method called `jobs.Push` to push the job to the worker. Moreover, you may push your job message directly into the queue driver, e.g. rabbitmq and it'll be consumed by RR.

> Also, I store some data with rr key-value plugin by php, how do I access it in golang?

Just connect to your storage and read the data.

**Reply by ryan053857** [reply]
Actually I am new to golang, can you give me the sample code? The main reason is that I don't know how to get examples of other plugin instance in the plugin. There is no detailed explanation in the document, thank you very much

**Reply by ryan053857** [reply]
Can you help me?

**Comment by ryan053857** [comment]
I already know how to use it, okay
